cronexpr.go

```diff
@@ -160,78 +160,122 @@ func (expr *Expression) Next(fromTime time.Time) time.Time {
 		return fromTime
 	}
 
-	// Since expr.nextSecond()-expr.nextMonth() expects that the
-	// supplied time stamp is a perfect match to the underlying cron
-	// expression, and since this function is an entry point where `fromTime`
-	// does not necessarily matches the underlying cron expression,
-	// we first need to ensure supplied time stamp matches
-	// the cron expression. If not, this means the supplied time
-	// stamp falls in between matching time stamps, thus we move
-	// to closest future matching immediately upon encountering a mismatching
-	// time stamp.
-
-	// year
-	v := fromTime.Year()
-	i := sort.SearchInts(expr.yearList, v)
-	if i == len(expr.yearList) {
+	loc := fromTime.Location()
+	t := fromTime.Add(time.Second - time.Duration(fromTime.Nanosecond())*time.Nanosecond)
```

Comment:
could truncate to time.Nanosecond here i guess

Reply:
Well they are rounding up - so I've moved the logic to a method explaining it. 
All hell breaks loose if I truncate to nanosec.